src/node_url.cc

```diff
@@ -311,9 +311,9 @@ void Initialize(Local<Object> target,
                 Local<Value> unused,
                 Local<Context> context,
                 void* priv) {
-  SetMethod(context, target, "parse", Parse);
-  SetMethod(context, target, "updateUrl", UpdateUrl);
-  SetMethod(context, target, "formatUrl", FormatUrl);
+  SetMethodNoSideEffect(context, target, "parse", Parse);
+  SetMethodNoSideEffect(context, target, "updateUrl", UpdateUrl);
+  SetMethodNoSideEffect(context, target, "formatUrl", FormatUrl);
```

Comment:
Why are these part of this PR? `Parse` and `UpdateUrl` absolutely have side effects: they call the provided callback. I don't see any side effects in `FormatUrl`, but changing it as part of this PR is still a bit awkward.

Reply:
I'll put this to a different pull request. You're right.